Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -109,6 +109,18 @@ static bool didDeflateFail(int result)
 }
 
 ExceptionOr<Ref<JSC::ArrayBuffer>> CompressionStreamEncoder::compress(std::span<const uint8_t> input)
+{
+#if PLATFORM(COCOA)
+    if (m_format == Formats::CompressionFormat::Brotli)
+        return compressAppleCompressionFramework(input);
+    else
+        return compressZlib(input);
```

Comment:
Can we use the compression framework for zlib too?  That would make our abstractions better organized

Reply:
Apple's Compression Framework did not fully support streaming of either `deflate`, `deflate-raw`, or `gzip`.

My memory is a bit fuzzy at the moment on which algorithm specifically was missing.